aten/src/ATen/native/cuda/ScanUtils.cuh

```diff
@@ -330,35 +357,36 @@ __device__ void tensor_kernel_scan_innermost_dim_impl(T* row_buf, T *tgt_, const
 
 template <
     typename T,
-    int num_threads_x,
-    int num_threads_y,
+    int num_threads,
     class BinaryFunction>
 __global__ typename std::enable_if<!c10::is_complex<T>::value, void>::type
 tensor_kernel_scan_innermost_dim(
     T* tgt_,
     const T* src_,
     const uint32_t num_rows,
     const uint32_t row_size,
+    const uint32_t log_num_threads_x,
     T init,
     BinaryFunction binary_op) {
-  __shared__ T sbuf[num_threads_y][2 * num_threads_x];
-  T* row_buf = sbuf[threadIdx.y];
+  __shared__ T sbuf[num_threads * 2];
```

Comment:
Yes, I can do that. In line with this, is there an internal function that determines the maximum number of threads automatically rather than manually specified as 512?

Reply:
You could query `maxThreadsPerBlock` https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__EXECUTION.html#group__CUDART__EXECUTION_1g19b1696533069c03f646e2ce2beacc00, however, that's not necessarily the best perf, because launching with maxThreads thus returned may result in lower occupancy. `512` is usually a good middle ground.

Reply:
Yes, I can set it dynamically. Is there any internal function to get the maximum number of threads instead of manually set it to 512?